src/explorer/ExplorationLauncher.cs

```diff
@@ -26,41 +26,38 @@ public ExplorationLauncher(IContainer rootContainer)
         /// An action to add and configure the components to use in this exploration.
         /// </param>
         /// <returns>The running Task.</returns>
-        public static async Task Explore(
+        public static Exploration Explore(
             INestedContainer scope,
             ExplorerContext ctx,
             DConnection conn,
             Action<ExplorationConfig> componentConfiguration)
         {
             // Configure a new Exploration
-            var exploration = Exploration.Configure(scope, _ =>
+            return Exploration.Configure(scope, _ =>
             {
                 _.UseConnection(conn);
                 _.UseContext(ctx);
                 _.Compose(componentConfiguration);
             });
-
-            // Run and await completion of all components
-            await exploration.Completion;
```

Comment:
I don't understand why you removed this line.
Considering this change, would it make sense to rename both methods in this class? Maybe something on the lines of `CreateExploration` will more accurately describe what these methods do now. And then it will make sense to rename the class also, e.g. something like `ExplorationFactory`, although `ExplorationLauncher` sounds very nice .

Reply:
I removed the line simply because the method now returns an `Exploration`, not a `Task` (because we need to encapsulate the `MetricsPublisher`). I think this makes more sense anyway - the caller should decide when and where to await the completion. 

I also considered renaming but launcher sounds nicer than factory... and also I think it makes it clear that the returned exploration is already running in the background. 

Btw. I'm actually not sure how/why the task is already running. I was expecting to need to call `Task.Run(exploration.Completion)` or `await exploration.Completion` somewhere but this wasn't necessary...